docs/lib/graphqlapi/fragments/ios/query-data.md

```diff
@@ -121,3 +119,91 @@ func listTodos() -> AnyCancellable {
 </amplify-block>
 
 </amplify-block-switcher>
+
+### List subsequent pages of items
+
+For large data sets, you'll need to paginate through the results. After receiving the first page of results, you can check if there is a subsequent page and obtain the next page.
+
+```swift
+var todos: [Todo] = []
+var currentPage: List<Todo>?
+
+func listFirstPage() {
+    let todo = Todo.keys
+    let predicate = todo.name == "my first todo" && todo.description == "todo description"
+    Amplify.API.query(request: .paginatedList(Todo.self, where: predicate, limit: 1000)) { event in
+        switch event {
+        case .success(let result):
+            switch result {
+            case .success(let todos):
+                print("Successfully retrieved list of todos: \(todos)")
+                self.currentPage = todos
+                self.todos.append(contentsOf: todos)
+            case .failure(let error):
+                print("Got failed result with \(error.errorDescription)")
+            }
+        case .failure(let error):
+            print("Got failed event with error \(error)")
+        }
+    }
+}
+
+func listNextPage() {
+    if let current = currentPage, current.hasNextPage() {
+        current.getNextPage { result in
+            switch result {
+            case .success(let todos):
+                self.todos.append(contentsOf: todos)
+                self.currentPage = todos
+            case .failure(let coreError):
+                print("Failed to get next page \(coreError)")
+            }
+        }
+    }
+}
+```
+
+## List all pages
+
+If you want to get all pages, retrieve the subsequent page when you have successfully retrieved the first or next page.
```

Comment:
If the code change is as small as I think it is, you can make understanding clearer (at the expense of copy/pasteability) by just calling out the one added line in the `listAllPages()` method, and then listing the code for the recursive method. Something like 

    If you want to get all pages, you can adapt the above code to recursively call `getNextPage()` after you retrieve the first page of results.

    - Create a new method named `listAllTodos()`, with the same content as the `listFirstPage()` method above.
    - After you append the first page of results to `self.todos`, add a call to `self.listNextPageRecursively()`:
        ```diff
         self.currentPage = todos
         self.todos.append(contentsOf: todos)
        +self.listNextPageRecursively()
        ```
    - Add the new `listNextPageRecursively()` method:
        ```swift
        func listNextPageRecursively() {
            if let current = currentPage, current.hasNextPage() {
                current.getNextPage { result in
                    switch result {
                    case .success(let todos):
                        self.todos.append(contentsOf: todos)
                        self.currentPage = todos
                        self.listNextPageRecursively()
                    case .failure(let coreError):
                        print("Failed to get next page \(coreError)")
                    }
                }
            }
        }
        ```

(Or with appropriate adaptations to use Combine as @jamesonwilliams suggested)

Reply:
the latest improvement to this describes the steps and then provides the final code snippet to allow copypasta. I think this provides the benefit of explaining what was done in the code and having the entire code snippet, i'm able to mark the step in the code like

1. Update the above method `listFirstPage()` to `listAllPages()` 
```
func listAllPages() { // 1. Updated from `listFirstpage()`
```